lib/verify.js

```diff
@@ -44,14 +44,14 @@ module.exports = {
    * that was returned from `parse()`.
    *
    * @param {Object} parsedSignature the object you got from `parse`.
-   * @param {String} secret HMAC shared secret.
+   * @param {String} or {Buffer} secret HMAC shared secret.
    * @return {Boolean} true if valid, false otherwise.
    * @throws {TypeError} if you pass in bad arguments.
    * @throws {InvalidAlgorithmError}
    */
   verifyHMAC: function verifyHMAC(parsedSignature, secret) {
     assert.object(parsedSignature, 'parsedHMAC');
-    assert.string(secret, 'secret');
+    assert(typeof (secret) === 'string' || secret instanceof Buffer);
```

Comment:
Is this a Node-ism?  As a novice, seeing TWO DIFFERENT types of comparison seems... weird.  Can't have `typeof (secret) === 'Buffer'` on the 2nd part of this assertion?

Reply:
If `secret` is a buffer, `typeof (secret) === 'object'`. For string verifications, from what I can tell, the usual validation is `typeof (variable) === 'string'`. I don't know if there's a better way to do that -- I could't find anything that seemed obvious.

Reply:
<sigh> Thanks Node.

Reply:
Probaby should have used `Buffer.isBuffer(secret)` here -- using `instanceof` can break in interesting ways if `Buffer` is a polyfill or being manipulated by something else (happens a lot in browser environments). It's generally a good rule to avoid ever using `instanceof` if you have any alternative at all.

Reply:
I can do a second PR before we make a release to fix that if you'd be willing to review it :)